From 9b55c3d5385bdc9724ea5cdbbac8d0f1dbe6bcbf Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 14 Mar 2018 10:01:24 +0900
Subject: [PATCH 683/909] arm64: dts: r8a7796-m3ulcb: Fix souce clock for DU

commit 392e61f1bf6029b8a160260b7b96b60fd16bee97 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch changes the clock for DU, supply the initial value of
versaclock directly to the DU.

The initial value of versaclock out3 is 33 MHz.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7796-m3ulcb.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb.dts b/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb.dts
index 3e6a4e111b38..c962301b5f34 100644
--- a/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7796-m3ulcb.dts
@@ -70,6 +70,13 @@
 	vspm_if {
 		compatible = "renesas,vspm_if";
 	};
+
+	versaclock5_out3: versaclk-3 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* Initial value of versaclock out3 */
+		clock-frequency = <33000000>;
+	};
 };
 
 &a53_0 {
@@ -93,7 +100,7 @@
 		 <&cpg CPG_MOD 723>,
 		 <&cpg CPG_MOD 722>,
 		 <&versaclock5 1>,
-		 <&versaclock5 3>,
+		 <&versaclock5_out3>,
 		 <&versaclock5 2>;
 	clock-names = "du.0", "du.1", "du.2",
 		      "dclkin.0", "dclkin.1", "dclkin.2";
-- 
2.17.1

